src/ChunkSender.cpp

```diff
@@ -247,64 +217,62 @@ void cChunkSender::Execute(void)
 
 
 
-void cChunkSender::SendChunk(int a_ChunkX, int a_ChunkZ, cClientHandle * a_Client)
+void cChunkSender::SendChunk(int a_ChunkX, int a_ChunkZ, std::unordered_set<cClientHandle *> a_Clients)
 {
-	ASSERT(m_World != nullptr);
 	
 	// Ask the client if it still wants the chunk:
-	if ((a_Client != nullptr) && !a_Client->WantsSendChunk(a_ChunkX, a_ChunkZ))
+	for(auto itr = a_Clients.begin(); itr != a_Clients.end();)
 	{
-		return;
+		if(!(*itr)->WantsSendChunk(a_ChunkX, a_ChunkZ))
+		{
+			auto toremove = itr;
+			itr++;
+			a_Clients.erase(toremove);
+		}
+		else
+		{
+			itr++;
```

Comment:
Apparently, you can erase normally when iterating a set or map as all implementations ensure order is unchanged anyway (as does C++14).